docs/maui/behaviors/numeric-validation-behavior.md

```diff
@@ -0,0 +1,142 @@
+             xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+             x:Class="CommunityToolkit.Maui.Sample.Pages.Converters.NumericValidationBehaviorPage">
+
+    <ContentPage.Resources>
+        <Style x:Key="InvalidEntryStyle" TargetType="Entry">
+            <Setter Property="TextColor" Value="Red" />
+        </Style>
+        <Style x:Key="ValidEntryStyle" TargetType="Entry">
+            <Setter Property="TextColor" Value="Green" />
+        </Style>
+    </ContentPage.Resources>
+
+    <Entry Keyboard="Numeric">
+        <Entry.Behaviors>
+            <toolkit:NumericValidationBehavior 
+                InvalidStyle="{StaticResource InvalidEntryStyle}"
+                ValidStyle="{StaticResource ValidEntryStyle}"
+                Flags="ValidateOnValueChanged"
+                MinimumValue="1.0"
+                MaximumValue="100.0"
+                MaximumDecimalPlaces="2" />
+        </Entry.Behaviors>
+    </Entry>
+
+</ContentPage>
+```
+
+### C#
+
+The `NumericValidationBehavior` can be used as follows in C#:
+
+```csharp
+class NumericValidationBehaviorPage : ContentPage
+{
+    public NumericValidationBehaviorPage()
+    {
+        var entry = new Entry
+        {
+            Keyboard = Keyboard.Numeric
+        };
+
+        var validStyle = new Style(typeof(Entry));
+        validStyle.Setters.Add(new Setter
+        {
+            Property = Entry.TextColorProperty,
+            Value = Colors.Green
+        });
+
+        var invalidStyle = new Style(typeof(Entry));
+        invalidStyle.Setters.Add(new Setter
+        {
+            Property = Entry.TextColorProperty,
+            Value = Colors.Red
+        });
+
+        var numericValidationBehavior = new NumericValidationBehavior
+        {
+            InvalidStyle = invalidStyle,
+            ValidStyle = validStyle,
+            Flags = ValidationFlags.ValidateOnValueChanged,
+            MinimumValue = 1.0,
+            MaximumValue = 100.0,
+            MaximumDecimalPlaces = 2
+        };
+
+        entry.Behaviors.Add(numericValidationBehavior);
+
+        Content = entry;
+    }
+}
+```
+
+### C# Markup
+
+Our [`CommunityToolkit.Maui.Markup`](../markup/markup.md) package provides a much more concise way to use this converter in C#.
+
+```csharp
+using CommunityToolkit.Maui.Markup;
+
+class NumericValidationBehaviorPage : ContentPage
+{
+    public NumericValidationBehaviorPage()
+    {
+        Content = new Entry
+        {
+            Keyboard = Keyboard.Numeric
+        }.Behaviors(
+            new NumericValidationBehavior
+            {
+                InvalidStyle = new Style<Entry>(Entry.TextColorProperty, Colors.Red),
+                ValidStyle = new Style<Entry>(Entry.TextColorProperty, Colors.Green),
+                Flags = ValidationFlags.ValidateOnValueChanged,
+                MinimumValue = 1.0,
+                MaximumValue = 100.0,
+                MaximumDecimalPlaces = 2
+            });
+    }
```

Comment:
Let's keep `NumericValidationBehavior` on the same line as `.Behaviors(` since it looks a bit out of place floating below it on the next line:

```suggestion
        Content = new Entry
        {
            Keyboard = Keyboard.Numeric
        }.Behaviors(new NumericValidationBehavior
        {
            InvalidStyle = new Style<Entry>(Entry.TextColorProperty, Colors.Red),
            ValidStyle = new Style<Entry>(Entry.TextColorProperty, Colors.Green),
            Flags = ValidationFlags.ValidateOnValueChanged,
            MinimumValue = 1.0,
            MaximumValue = 100.0,
            MaximumDecimalPlaces = 2
        });
    }
```